chalicelib_smaht/checks/helpers/wfr_utils.py

```diff
@@ -1,875 +1,40 @@
-        check.status = 'WARN'
-    if missing_meta_changes:
-        check.allow_action = True
-        check.summary = 'Some files are missing runs'
-        check.brief_output.append(str(len(missing_meta_changes)) + ' files have successful run but no qc/extra file')
-        check.full_output['files_without_changes'] = missing_meta_changes
-        check.status = 'WARN'
-    check.summary = check.summary.strip()
-    if not check.brief_output:
-        check.brief_output = ['All Good!']
-    return check
-
-
-def is_there_my_qc_metric(file_meta, qc_metric_name, my_auth):
-    if not file_meta.get('quality_metric'):
-        return False
-    qc_results = ff_utils.get_metadata(file_meta['quality_metric']['uuid'], key=my_auth)
-    if qc_results['display_title'].startswith('QualityMetricQclist'):
-        if not qc_results.get('qc_list'):
-            return False
-        for qc in qc_results['qc_list']:
-            if qc_metric_name not in qc['value']['display_title']:
-                return False
-    else:
-        if qc_metric_name not in qc_results['display_title']:
-            return False
-    return True
-
-
-def fetch_wfr_associated(wfr_info):
-    """Given wfr embedded frame, find associated output files and qcs"""
-    wfr_as_list = []
-    wfr_as_list.append(wfr_info['uuid'])
-    if wfr_info.get('output_files'):
-        for o in wfr_info['output_files']:
-            if o.get('value'):
-                wfr_as_list.append(o['value']['uuid'])
-            elif o.get('value_qc'):
-                wfr_as_list.append(o['value_qc']['uuid'])
-    if wfr_info.get('output_quality_metrics'):
-        for qc in wfr_info['output_quality_metrics']:
-            if qc.get('value'):
-                wfr_as_list.append(qc['value']['uuid'])
-    if wfr_info.get('quality_metric'):
-        wfr_as_list.append(wfr_info['quality_metric']['uuid'])
-    return list(set(wfr_as_list))
+from packaging import version
+
+
+def get_md5_mwfrs_for_file(my_auth, file_uuid, md5_mwf_uuid):
+    query = f"/search/?type=MetaWorkflowRun&meta_workflow.uuid={md5_mwf_uuid}&input.files.file.uuid={file_uuid}"
+    return ff_utils.search_metadata(query, key=my_auth)
+
+
+def get_latest_md5_mwf(my_auth):
+    # We assume that md5 MetaWorkflows have name "md5". We have a similar strong assumption in Tibanna.
+    query = f"/search/?type=MetaWorkflow&name=md5"
+    search_results = ff_utils.search_metadata(query, key=my_auth)
+    
+    if len(search_results) == 0:
+        return None
+    
+    latest_md5_item = search_results[0]
+    if len(search_results) == 1:
+        return latest_md5_item
+    
+    # There are multiple MWFs. Get the latest version
+    for search_result in search_results:
+        if version.parse(latest_md5_item["version"]) < version.parse(search_result["version"]):
+            latest_md5_item = search_result
+    return latest_md5_item
+
+# Unused right now - maybe useful later
+def pageinate_list(list, page_size):
```

Comment:
Think this should be "paginate" for the spelling (?).
```suggestion
def paginate_list(list, page_size):
```